util/reggen/data.py

```diff
@@ -230,10 +232,12 @@ def __init__(self):
         self.addr_width = 12
         self.base_addr = 0
         self.name = ""
+        self.instance = ""
         self.hier_path = ""
         self.regs = []
         self.wins = []
         self.blocks = []
+        self.ips = []
```

Comment:
Logically, 'blocks' within a parent 'block' was indeed meant to represent an IP. There is no need to add extra 'ips' list - please reuse 'blocks' for all corresponding changes.

Reply:
`ips` and `blocks` are used differently. `ips` is unique module, `blocks` is for the instance. for example, we have 2 EDN in the top. `ips` contains 1 EDN object, but `blocks` has 2 EDN objects.
for generating the reg class of the IP, we use
```
% for i in block.ips:
${construct_classes(i)}
% endfor
```
for creating instances of the reg class, we use
```
% for b in block.blocks:
    rand ${gen_dv.bcname(b)} ${b.instance};
% endfor
```

we can extract `ips` from `blocks` in `uvm_reg.sv.tpl` but `ips` is easily obtained at topgen, so I have both in the `Block`

Reply:
This can be removed now, right?

Reply:
oops, forgot to remove this. Done. thanks.